handsontable/handsontable
# Add hook `beforeFilterValueOptionsShow ` for customize options order in fiter dropdown menu

### Context
Provide the ability to adjust options order in filter dropdown menu.

### How has this been tested?
I have added two tests in `filters.spec.js` hooks section
- Verify new `beforeFilterValueOptionsShow` hook will be called before when dropdown is show.
- Verify hook handler could modify dropdown menus order.

Also all unit and e2e tests have passed.

### Types of changes


### Related issue(s):
* [Changing order of “filter by value” data](https://forum.handsontable.com/t/changing-order-of-filter-by-value-data/3260)
* [Add case sensitive options for filters formulas](https://github.com/handsontable/handsontable/discussions/7707)
* [Is it possible customize options order in filter menu?](https://forum.handsontable.com/t/is-it-possible-customize-options-order-in-filter-menu/5329)

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:


- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
